packages/pwa/app/commerce-api/index.js

```diff
@@ -117,44 +110,39 @@ class CommerceAPI {
-                                    // Ensure we don't send the pseudo locale.
-                                    if (locale === 'en-XB') {
-                                        sendLocale = false
-                                    }
-
-                                    // Inject the locale and currency to the API call via it's parameters.
-                                    //
-                                    // NOTE: The commerce sdk isomorphic will complain if you pass parameters to
-                                    // it that it doesn't expect, this is why we only add the locale and currency
-                                    // to some of the API calls.
-                                    // We use the default locale for the API calls when running the app using the
-                                    // pseudo locale 'en-XB'.
-                                    newArgs[0].parameters = {
-                                        ...newArgs[0].parameters,
-                                        ...(sendLocale ? {locale} : {}),
-                                        ...(sendCurrency ? {currency} : {})
-                                    }
+                                // Inject the locale and currency to the API call via its parameters.
+                                //
+                                // NOTE: The commerce sdk isomorphic will complain if you pass parameters to
+                                // it that it doesn't expect, this is why we only add the locale and currency
+                                // to some of the API calls.
+
+                                // By default we send the locale param and don't send the currency param.
+                                const {sendLocale = true, sendCurrency = false} = apiConfigs[key]
+
+                                const sendLocaleForCurrentProp = Array.isArray(sendLocale)
+                                    ? sendLocale.includes(prop)
+                                    : !!sendLocale
+                                const sendCurrencyForCurrentProp = Array.isArray(sendCurrency)
+                                    ? sendCurrency.includes(prop)
+                                    : !!sendCurrency
+
+                                const {locale: _l, currency: _c, ...rest} =
+                                    fetchOptions.parameters || {}
+
+                                fetchOptions.parameters = {
+                                    ...rest,
+                                    ...(sendLocaleForCurrentProp ? {locale} : {}),
+                                    ...(sendCurrencyForCurrentProp ? {currency} : {})
+                                }
 
+                                return self.willSendRequest(prop, ...args).then((newArgs) => {
                                     return obj[prop](...newArgs).then((res) =>
                                         self.didReceiveResponse(res, newArgs)
                                     )
```

Comment:
Some refactoring is done to move the locale/currency injection earlier. So that `willSendRequest` and `didReceiveResponse` are closer together for a more readable code.